src/main/java/com/google/impactdashboard/server/api_utilities/LogRetriever.java

```diff
@@ -0,0 +1,109 @@
+package com.google.impactdashboard.server.api_utilities;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.audit.AuditLog;
+import com.google.cloud.logging.v2.LoggingClient;
+import com.google.cloud.logging.v2.LoggingClient.ListLogEntriesPagedResponse;
+import com.google.cloud.logging.v2.LoggingSettings;
+import com.google.cloud.logging.v2.stub.LoggingServiceV2StubSettings;
+import com.google.logging.v2.ListLogEntriesRequest;
+import com.google.logging.v2.LogEntry;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Value;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+
+/** Class that handles all the retrieval of logs stored on the cloud logging API. */
+public class LogRetriever {
+
+  /** Cloud Logging client used to retrieve Audit logs and Recommendation logs */
+  private LoggingClient logger;
+
+  /**
+   * Static factory method for creating a new LogRetriever with a new instance of Logging client.
+   * @return A new instance of {@code LogRetriever}
+   */
+  public static LogRetriever create() throws IOException{
+    // Change logging client initialization to settings
+    LoggingServiceV2StubSettings stub = LoggingServiceV2StubSettings.newBuilder()
+        .setCredentialsProvider(() -> {
+          GoogleCredentials credentials;
+          try {
+            credentials = GoogleCredentials
+                // The path will be changed to use the constants class when it is merged into main
+                .fromStream(new FileInputStream("/usr/local/google/home/ionis/Documents/credentials.json"));
```

Comment:
same comment on configuration file.

Reply:
That's a good idea. I will work on the centralized class in the next updater PR, but I will fix the configuration file issue.